### PR TITLE
[ENT-644] Remove account-level & `EnterpriseCourseEnrollment` consent.

### PR DESCRIPTION
**Pull request by UmanShahzad**
**Description:** This removes any remaining traces of account-level consent and consent for courses tracked by the `EnterpriseCourseEnrollment` model.

**JIRA:** [ENT-644](https://openedx.atlassian.net/browse/ENT-644)

**Merge deadline:** N/A

**Testing instructions:**

Since account-level consent and course-level consent through `EnterpriseCourseEnrollment` were both deprecated in favor of `DataSharingConsent`, this should not have any effect on functionality. However, to be safe, it may be recommended to try out our data sharing consent flows to make sure everything still works as intended.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @UmanShahzad! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=192&repo=edx%2Fedx-enterprise"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: enterprise/decorators.py [hunk 1]
```diff
@@ -17,58 +15,6 @@
...
-        """
-        @wraps(func)
-        def wrapper(*args, **kwargs):
-            """
-            Wrap the function.
-            """
-            warnings.simplefilter('ignore', warning)
-            return func(*args, **kwargs)
-        return wrapper
-    return decorator
```
Comment (UmanShahzad) [inline enterprise/decorators.py]: I don't mind keeping them but they're unused, and deleting code is fun :]

File: tests/test_enterprise/api/test_views.py [hunk 2]
```diff
@@ -220,33 +185,28 @@ def test_enterprise_learner_entitlements(
...
-        Arguments:
-            enable_data_sharing_consent (bool): True if enterprise customer enables data sharing consent,
-                False it does not.
-            enforce_data_sharing_consent (str): string for the location at which enterprise customer enforces
-                data sharing consent, possible values are 'at_enrollment' and 'externally_managed'.
-            learner_consent_state (str): string containing the state of learner consent on data sharing,
-                possible values are 'not_set', 'enabled' and 'disabled'.
-            entitlements (list): A list of integers pointing to voucher ids generated in E-Commerce CAT tool.
-            expected_json (dict): A dict with structure and values corresponding to
-                the expected json from API endpoint.
```
Comment (UmanShahzad) [inline tests/test_enterprise/api/test_views.py]: I didn't think it was useful to maintain an argument list doc for a test function. The names should speak for themselves.

File: tests/test_enterprise/api/test_views.py [hunk 3]
```diff
@@ -220,33 +185,28 @@ def test_enterprise_learner_entitlements(
...
         """
-        user_id = 1
+        user_id = self.user.id + 1
```
Comment (UmanShahzad) [inline tests/test_enterprise/api/test_views.py]: Prevent conflicts => less flaky tests.

File: tests/test_enterprise/api/test_views.py [hunk 4]
```diff
@@ -220,33 +185,28 @@ def test_enterprise_learner_entitlements(
...
+            enterprise_customer=enterprise_customer,
+            granted=learner_consent_state,
         )
         for entitlement in entitlements:
             factories.EnterpriseCustomerEntitlementFactory(
                 enterprise_customer=enterprise_customer,
                 entitlement_id=entitlement,
             )
-        url = reverse('enterprise-learner-entitlements', (user_id, ))
+        url = reverse('enterprise-learner-entitlements', (ecu.id, ))
```
Comment (UmanShahzad) [inline tests/test_enterprise/api/test_views.py]: The ID being used here is for the learner instance, not the user instance.

File: tests/test_migrations.py [hunk 5]
```diff
@@ -77,97 +70,3 @@ def model_label(self):
...
-            granted=False,
-        )
-
-        self.enterprise_course_enrollment.consent_granted = True
-        self.enterprise_course_enrollment.save()
-
-        self.migrate_to_dest()
-
-        data_sharing_consent = DataSharingConsent.objects.all().first()
-        assert data_sharing_consent.granted
```
Comment (UmanShahzad) [inline tests/test_migrations.py]: Although the migration this was testing obviously still exists, there should be no issue for new users of edx-enterprise, so there's no need to test that code anymore.

File: enterprise/decorators.py [hunk 6]
```diff
@@ -17,58 +15,6 @@
 from six.moves.urllib.parse import parse_qs, urlencode, urlparse, urlunparse  # pylint: disable=import-error
 
 
-def deprecated(extra):
```
Comment (brittneyexline) [inline enterprise/decorators.py]: this isn't specific to the old consent stuff although it brought this up as being useful - i would argue we should keep this because i'm sure we'll be deprecating other stuff. unless there is another way we should be doing that going forward?

Comment (UmanShahzad) [reply enterprise/decorators.py]: Hmm, I think the decorators did help me remember what it was that needed to be deleted (that's why I put 'em in the first place -- and to bug us in the test logs into doing so). But I think now that it's unused, I'll have to add unit tests for coverage D: But no worries I'll do that :+1:

**Comment by UmanShahzad** [comment]
@brittneyexline It is not deployed yet, no. I'll do that early Monday my time along with #190.

**Comment by UmanShahzad** [comment]
This is now on the sandbox.

File: enterprise/decorators.py (hunk 6 above)
Comment (UmanShahzad) [reply enterprise/decorators.py]: @brittneyexline I've added these back and added unit tests.

**Comment by UmanShahzad** [comment]
Note that Zeshan has tested this on the sandbox and found no regressions.

File: tests/test_enterprise/api/test_views.py [hunk 7]
```diff
@@ -220,33 +185,28 @@ def test_enterprise_learner_entitlements(
...
-        factories.UserDataSharingConsentAuditFactory(
-            user__id=user_id,
-            user__enterprise_customer=enterprise_customer,
-            state=learner_consent_state,
+        user = factories.UserFactory(id=user_id)
+        ecu = factories.EnterpriseCustomerUserFactory(
+            user_id=user.id,
+            enterprise_customer=enterprise_customer,
+        )
+        factories.DataSharingConsentFactory(
```
Comment (haikuginger) [inline tests/test_enterprise/api/test_views.py]: Is this still doing something meaningful? Why aren't we passing a course ID (which should be required for this model), and why is the result working without a specific course ID? I might be missing something here.

Comment (UmanShahzad) [reply tests/test_enterprise/api/test_views.py]: The endpoint's function attempts to [get entitlements](https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L220) which then only [looks to see](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L417) if any DSC records exist between the learner and the customer, and if one does and DSC is enforced, the logic says consent isn't required.

I don't know if this is how DSC requirements for entitlements should actually be working, so please let me know and we'd have to fix that fast.

Comment (haikuginger) [reply tests/test_enterprise/api/test_views.py]: @douglashall, thoughts? I think we might have missed an edge case here; I think the appropriate fix would be to mark consent as required on the entitlements based _solely_ on the Enterprise setting, and then put logic in ecom to check for actual consent on the relevant courses. Or, you could even modify the existing consent views to redirect the user back to the success URL if consent straight-up isn't required.

Comment (douglashall) [reply tests/test_enterprise/api/test_views.py]: Is this something that would break our current coupon-voucher-based implementation of auto-discounts for enterprise learners? If so, we should fix it. Even though eventually, I think the entitlements endpoint can be removed entirely once we migrate over to using Enterprise Offers for the auto-enterprise-discount.

Comment (haikuginger) [reply tests/test_enterprise/api/test_views.py]: @douglashall I think it might. @UmanShahzad, see my thoughts on the implementation above.

Comment (UmanShahzad) [reply tests/test_enterprise/api/test_views.py]: @haikuginger @douglashall I will look into this, but I'm a little time-strapped with the programs landing page & the endpoint stuff. This issue was of course quite unexpected. I might have to ask @itsjeyd to help if I realize this may be a little too much for me alone right now. I'll let you guys know by tomorrow morning.

Comment (UmanShahzad) [reply tests/test_enterprise/api/test_views.py]: See https://openedx.atlassian.net/browse/ENT-644?focusedCommentId=286220&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286220 for further discussion on this.

**Comment by UmanShahzad** [comment]
Rebased on `master` from 609666adac4058b37dcf1bbebf8b2009e729f3bb.

**Comment by UmanShahzad** [comment]
Ugh, coverage is failing due to `master`. I'll try and add something to this PR.

**Comment by UmanShahzad** [comment]
Ok, fixed coverage from `master`.

File: tests/test_utilities.py [hunk 8]
```diff
@@ -912,21 +904,38 @@ def test_ungettext_min_max(self, singular, plural, range_text, min_val, max_val,
...
+        """
+        site = SiteFactory()
+        site.configuration = mock.MagicMock(get_value=mock.MagicMock(return_value='value'))
+        assert utils.get_configuration_value_for_site(site, 'key', 'default') == 'value'
+
+    def test_get_configuration_value_for_site_without_configuration(self):
+        """
+        ``get_configuration_value_for_site`` returns the default because of no site configuration.
+        """
+        assert utils.get_configuration_value_for_site(SiteFactory(), 'key', 'default') == 'default'
```
Comment (douglashall) [inline tests/test_utilities.py]: FYI @tuchfarber

**Comment by UmanShahzad** [comment]
Rebased on `master` from d248130e29f359cf54d863fa0182dc2b92f023f5.
